src/Interpreters/Streaming/WindowCommon.h

```diff
@@ -0,0 +1,122 @@
+#pragma once
+
+#include <Core/ColumnWithTypeAndName.h>
+#include <Parsers/IAST_fwd.h>
+#include <Parsers/ASTFunction.h>
+#include <Common/IntervalKind.h>
+
+
+namespace DB
+{
+
+namespace Streaming
+{
+
+struct WindowInterval
+{
+    Int64 interval = 0;
+    IntervalKind::Kind unit = IntervalKind::Second;
+
+    operator bool() const { return interval != 0; }
+};
+
+void checkIntervalAST(const ASTPtr & ast, const String & msg = "Invalid interval");
+void extractInterval(const ASTFunction * ast, Int64 & interval, IntervalKind::Kind & kind);
```

Comment:
I meant that this function is only used by the `WindowInterval extractInterval(const ASTFunction * ast)` function, so maybe this can be eliminated.

> if you prefer return a std::pair, yes we can do that. 

I think instead of `std::pair` a `WindowInternal` is better choice. However than this function would became exactly the same as the function mentioned above. That's why I was wondering whether we could move the implementation into the function below and remove this function.